tests/passw_hardening/test_passw_hardening.py

```diff
@@ -0,0 +1,606 @@
+                                        digit_class="false",
+                                        special_class='false')
+
+    review_one_policy_with_user(duthost, passw_hardening_ob, passw_test, passw_bad_test, passw_exp_error, PAM_PASSWORD_CONF_UPPER_LETTER_ONLY_EXPECTED)
+
+def test_passw_hardening_policies_special_class(duthosts, enum_rand_one_per_hwsku_hostname, clean_passw_policies, clean_passw_one_policy_user):
+    """ 
+        Test password hardening special class 
+        Good flow - set passw according the policy
+        Bad flow - set passw not according the policy and expecting to get an error
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    passw_test = 'nvipashar_'
+    passw_bad_test = 'no11spec'
+    passw_exp_error = 'BAD PASSWORD: is too simple'
+
+    # set new passw hardening policies values
+    passw_hardening_ob = PasswHardening(state='enabled',
+                                        expiration='0',
+                                        expiration_warning='0',
+                                        history='0',
+                                        len_min='1',
+                                        reject_user_passw_match='false',
+                                        lower_class='false',
+                                        upper_class='false',
+                                        digit_class="false",
+                                        special_class='true')
+
+    review_one_policy_with_user(duthost, passw_hardening_ob, passw_test, passw_bad_test, passw_exp_error, PAM_PASSWORD_CONF_SPECIAL_LETTER_ONLY_EXPECTED)
+
+def test_passw_hardening_policy_reject_user_passw_match(duthosts, enum_rand_one_per_hwsku_hostname, clean_passw_policies, clean_passw_one_policy_user):
+    """ 
+        Test password hardening reject user passw match
+        Bad flow - set passw not according the policy and expecting to get an error
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    passw_test = '19892022'
+    passw_bad_test = USERNAME_ONE_POLICY
+    passw_exp_error='BAD PASSWORD: contains the user name in some form'
+    
+    # set new passw hardening policies values
+    passw_hardening_ob = PasswHardening(state='enabled',
+                                        expiration='0',
+                                        expiration_warning='0',
+                                        history='0',
+                                        len_min='1',
+                                        reject_user_passw_match='true',
+                                        lower_class='false',
+                                        upper_class='false',
+                                        digit_class="false",
+                                        special_class='false')
+
+    review_one_policy_with_user(duthost, passw_hardening_ob, passw_test, passw_bad_test, passw_exp_error, PAM_PASSWORD_CONF_REJECT_USER_PASSW_MATCH_EXPECTED)
+
```

Comment:
I have a question for this scenario, may be need a UT to cover it:

let's say the device only have 1 user ‘admin’, and then admin password expired, so is there any solution to login to device?

If not this will cause very big incident, we need change feature to prevent this happen, for example add a white list or make su user not limit by this feature.

Reply:
this case its not really related to password-hardening feature.
If someone forget the password there are different ways to recover from it and it could happen even without the feature.
The admin should know how to support it.